sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py

```diff
@@ -509,8 +509,7 @@ def create_or_update(
             # Make a copy of self._kwargs instead of contaminate the original one
             kwargs = dict(**self._kwargs)
             if hasattr(rest_job_resource.properties, "identity") and (
-                rest_job_resource.properties.identity is None
```

Comment:
for SparkJob, by default, we use user identity pass through. will removing this break it?

Reply:
I would prefer that to be defaulted in spark job class itself so that it gets passed along. The above check was adding it even when identity property was None

Reply:
got it. then we should make the change in the same release to avoid breaking. @DouglasXiaoMS

Reply:
I think we need to change here: when self.identity is None, set UserIdentity

https://github.com/Azure/azure-sdk-for-python/blob/5625581056184d46cbb539ea998a77931f90725b/sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/spark_job.py#L230

Reply:
@singankit , please hold on the merge (or at least the release of the new SDK) till the corresponding change is made in SpakJob

Reply:
Tony Ponwin Jeparatnam   is going to make the change in SparkJob